### PR TITLE
Drop JAXB from CertEnrollmentRequest

### PR DESCRIPTION
**Pull request by c-dorney**
Added DOM mappings for CertEnrollmentRequest

### CODE REVIEW

**Review by ckelleyRH — APPROVED** [review]
LGTM.

**Comment by ckelleyRH** [comment]
The PR could probably do with being renamed `Drop JAXB from CertEnrollmentRequest`, just noticed it has the same name as the branch.

File: base/common/src/main/java/com/netscape/certsrv/cert/CertEnrollmentRequest.java [hunk 1]
```diff
@@ -349,18 +360,157 @@ public boolean equals(Object obj) {
...
+        NodeList renewalList = certEnrollmentRequestElement.getElementsByTagName("Renewal");
+        if (renewalList.getLength() > 0) {
+            String value = renewalList.item(0).getTextContent();
+            certEnrollmentRequest.setRenewal(Boolean.valueOf(value));
+        }
+
+        NodeList inputList = certEnrollmentRequestElement.getElementsByTagName("Input");
+        for (int i = 0; i < inputList.getLength(); i++) {
+          Element inputElement = (Element) inputList.item(i);
+          ProfileInput profileInput = new ProfileInput();
```
Comment (edewata) [inline base/common/src/main/java/com/netscape/certsrv/cert/CertEnrollmentRequest.java:426]: The code inside the loop should be indented with 4 spaces.

**Review by edewata — APPROVED** [review]
It looks good in general, but I have some comments below:

* The DOM mapping doesn't include all fields in `CertEnrollmentRequest`, but since the CI passed this is probably fine for now. We can add any missing mapping later.
* The branch contains a merge commit. We usually don't use merge commits. Could you rebase the branch against `master` before merging?
* There is a formatting issue below.

Please address comments #2 and #3 if you could before merging. Thanks!

**Comment by c-dorney** [comment]
> * The DOM mapping doesn't include all fields in `CertEnrollmentRequest`, but since the CI passed this is probably fine for now. We can add any missing mapping later.

Nice spot! I will add those in now. 

> * The branch contains a merge commit. We usually don't use merge commits. Could you rebase the branch against `master` before merging?

Old habit of using merge to update my local branches, will use rebase going forward :smile: 

> * There is a formatting issue below.

Fixed :smile:

File: base/common/src/main/java/com/netscape/certsrv/cert/CertEnrollmentRequest.java [hunk 2]
```diff
@@ -349,18 +360,225 @@ public boolean equals(Object obj) {
...
+
+        for (ProfileOutput output: getOutputs()) {
+
+            certEnrollmentRequestElement.appendChild(output.toDOM(document));
+
+            NodeList nodes = certEnrollmentRequestElement.getElementsByTagName("profileOutput");
+            System.out.println(nodes.getLength());
+            for (int i=0; i < nodes.getLength(); i++) {
+                document.renameNode(nodes.item(i), null, "Output");
+            }
```
Comment (edewata) [inline base/common/src/main/java/com/netscape/certsrv/cert/CertEnrollmentRequest.java:424]: That should work, so feel free to merge (but remove the `println()` first).

If you want, you probably could refactor `ProfileOutput.toDOM()` like this:
```
public void toDOM(Document document, Element element) {
    element.setAttribute("id", id);
    ...
}

public Element toDOM(Document document) {
    Element poElement = document.createElement("profileOutput");
    toDOM(document, poElement);
    return poElement;
}
```
Then you can call it like this:
```
Element output = document.createElement("Output");
output.toDOM(document, output);
createEnrollmentRequestElement.appendChild(output);
```

Comment (c-dorney) [reply base/common/src/main/java/com/netscape/certsrv/cert/CertEnrollmentRequest.java]: Updated. Thanks that looks a lot cleaner!

**Pull request merged**